lib/activerecord/postgres_pub_sub/listener.rb

```diff
@@ -87,10 +91,11 @@ def empty_channel(connection)
 
       def wait_for_notify(connection)
         connection_pid = connection.raw_connection.backend_pid
-        event_result = connection.raw_connection.wait_for_notify(listen_timeout) do |_event, pid, payload|
+        event_result = connection.raw_connection.wait_for_notify(listen_timeout) do |notify_channel, pid, payload|
           if pid != connection_pid
             empty_channel(connection.raw_connection) if notify_only
-            yield(payload)
+
+            channels.size > 1 ? yield(payload, notify_channel) : yield(payload)
```

Comment:
Do we need this explicitly check, or can we yield both arguments regardless of the # of channels?

Reply:
Cool. I just too overly worried about that it is a spec changing for others legacy code.  fixed now.

Reply:
Thanks, @TimWei! Your point about specs is well-taken. What do you think about adding an explicit spec for the 1-arg version of `on_notify`, to ensure we don't break that behavior both here and going forward?

Reply:
I added a new context which using 1-arg version of `on_notify` with a test case.